builtins: implement ST_Point and ST_MakePoint

otan opened the pull request:
Resolves https://github.com/cockroachdb/cockroach/issues/48819
Resolves https://github.com/cockroachdb/cockroach/issues/48815

Release note (sql change): Implemented the ST_Point and ST_MakePoint
functions.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/49194)
<!-- Reviewable:end -->

otan committed:
builtins: implement ST_Point and ST_MakePoint

Release note (sql change): Implemented the ST_Point and ST_MakePoint
functions.

sumeerbhola reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> 

> Reviewed 4 of 4 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/49194#-:-M7d2COr2cWtmyPPRU9l:bg7igkv)* status: :shipit: complete! 1 of 0 LGTMs obtained



<!-- Sent from Reviewable.io -->

otan commented:
bors r=sumeerbhola
